zip-reduce-shielded_tx_fee.rst

```diff
@@ -0,0 +1,135 @@
+to 1000 zats.
+
+With reduced fees, it will be cheaper to transact on the Zcash network,
+while also inviting novel use cases for privacy
+preserving applications that would benefit from the Privacy,
+Security and Programmable Money aspects of the Zcash chain.
+
+
+Out of Scope for this Proposal
+============
+
+Discussion around transaction fees for T to T, T to Z or Z to T.
+
+
+Motivation
+============
+
+The default transaction fee presently is 0.0001 ZEC or 10,000 zats.
+With ZEC market price of $100, a user can send 10,000 transactions
+for 1 ZEC, that turns out to 1 cent per transaction and it rises
+with the increase in the price of ZEC.
+
+With increase in light wallet adoptions on mobile clients, many users
+will be new to the Zcash eco-system. And the fact that the
+transaction fees are paid by the sender (not the receiver) is
+new information to users who might use Zcash for e-commerce
+and app interactions that might result in several transactions each day.
+
+Privacy must not cost a premium. The cost of 10 shielded transactions
+buys 1GB of mobile data `in India today <https://www.cable.co.uk/mobiles/worldwide-data-pricing/>`.
+
+Zcash users must be able to do more with their ZEC balance
+than worry about paying the premium for shielded transactions.
+
+
+Requirements for consensus
+-------------------------
+
+The change to the conventional transaction fees must be undertaken sooner
+as it gets difficult to gain consensus with the growth in the network
+of wallets, exchanges, miners and third parties involved.
+
+The following parties need to be part of the consensus:
+
+* A guarantee from mining groups is required to include the lowered default fee transactions in the next block.
+* Wallet developers need to provide commitment to update the software to use the new fee.
+* Zcash documentation and community outreach must be undertaken to make the change known.
+
+
+Security and privacy considerations
+-----------
+
+Unique transaction fees can cause linkability within transactions,
+hence this change must be accepted by majority, if not all popular
+shielded transaction software providers before announcing the change.
+
+Varying/unique fees are bad for privacy, for the short term before blocks get full,
+it’s fine for everyone to use a constant fee. [#nathan-1]_
+
+Long term, the issue of fees needs to be re-visited as the blocks start getting consistently full.
+And the possibility of a DoS scenario becomes critical in nature. New ZIP with flexible fees [#ian-1]_
+along with scaling solutions need to be evaluated and applied.
+
+
+Activation
+============
+
+* The recommended fee of 0.00001 or 1000 zats must be activated at block 1046400 to coincide with the `Canopy upgrade. <https://z.cash/upgrade/canopy/>`
+* With a grace period of ~4 weeks (block 1080000) to upgrade to the reduced default transaction fee for zcashd and core clients used by exchanges & service providers.
```

Comment:
I have received feedback that this activation is too rapid because ECC commissioned a marketing video and wasn't aware of this ZIP change, so the user education video is now incorrect. This is a specific example of a general difficult problem of ecosystem coordination.

This ZIP carries a rationale that a quick deployment is important to improve adoption, because a longer postponement might make it harder to reach consensus:

> The change to the conventional transaction fees must be undertaken sooner
as it gets difficult to gain consensus with the growth in the network
of wallets, exchanges, miners and third parties involved.

In my personal opinion this is a correct motivation _but_ also the timescale is several months given the current pace of Zcash adoption. I personally advocate for _not changing_ this activation schedule because there is already momentum behind this schedule from the majority of shielded wallets and it matches up with Canopy.

However, I think we should also establish some norms/conventions about future ZIP schedules that include a longer "comment period" before a ZIP is "committed to deployment" and also a delay from "committed to deployment" to "feature activation". This deserves a completely separate discussion from this review. (This is very closely connected to the Network Upgrade Pipeline and the need to improve it.)

Reply:
Agree, the Timeline to lower the shielded transaction fees sneaked up too quick.
The Activation Schedule is now updated to:

> The recommended fee of 0.00001 or 1000 zats must start activation at block 1,080,000
> With a grace period of ~4 weeks (block 1,120,000) to upgrade to the reduced

This is in line with ZecWallet's implementation to activate the new lower default fee starting for block 1,080,000

[Zcash Forum Comment link](https://forum.zcashcommunity.com/t/zip-reduce-default-shielded-transaction-fee-to-1000-zats/37566/62?u=aiyadt)